File: android/src/main/java/com/davecoates/rnfirebasebridge/FirebaseBridgeEmailPasswordAuthProvider.java
package com.davecoates.rnfirebasebridge;

import com.facebook.react.bridge.*;
import com.google.firebase.auth.AuthCredential;
import com.google.firebase.auth.EmailAuthProvider;

/**
 * Created by hjumeau on 31/01/2017.
 */

public class FirebaseBridgeEmailPasswordAuthProvider extends ReactContextBaseJavaModule {

    public FirebaseBridgeEmailPasswordAuthProvider(ReactApplicationContext reactContext) {
        super(reactContext);
    }

    @Override
    public String getName() {
        return "FirebaseBridgeEmailPasswordAuthProvider";
    }

    @ReactMethod
    public void credential(String email, String password, Promise promise) {
        AuthCredential credential = EmailAuthProvider.getCredential(email, password);
        WritableMap data = Arguments.createMap();
        data.putString("id", FirebaseBridgeCredentialCache.addCredential(credential));
        data.putString("provider", credential.getProvider());
        promise.resolve(data);
    }
}

File: ios/FirebaseBridgeEmailPasswordAuthProvider.swift
//
//  FirebaseBridgeEmailPasswordAuthProvider.swift
//  ChenBoul
//
//  Created by Harry JUMEAU on 31/01/2017.
//  Copyright © 2017 Facebook. All rights reserved.
//

import Firebase

@objc(FirebaseBridgeEmailPasswordAuthProvider)
class FirebaseBridgeEmailPasswordAuthProvider : NSObject {
  
  var bridge: RCTBridge!
  
  override init() {
    super.init()
  }
  
  @objc func credential(email:String, password:String, resolver resolve: RCTPromiseResolveBlock, rejecter reject: RCTPromiseRejectBlock) {
    let credential = FIREmailPasswordAuthProvider.credentialWithEmail(email, password: password)
    resolve([
      "id": FirebaseBridgeCredentialCache.addCredential(credential),
      "provider": credential.provider,
      ])
  }
}
